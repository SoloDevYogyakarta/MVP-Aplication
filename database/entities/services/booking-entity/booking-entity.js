"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bookingEntity = exports.BookingEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class BookingEntity extends base_entity_1.BaseEntity {
}
exports.BookingEntity = BookingEntity;
exports.bookingEntity = entity_1.sequelize.define('BOOKING', {
    public_id: sequelize_1.DataTypes.STRING,
    username: sequelize_1.DataTypes.STRING,
    user_id: sequelize_1.DataTypes.STRING,
    day: sequelize_1.DataTypes.DATE,
    desc: sequelize_1.DataTypes.TEXT,
}, {
    tableName: 'BOOKING',
    schema: 'SERVICES',
});
//# sourceMappingURL=booking-entity.js.map