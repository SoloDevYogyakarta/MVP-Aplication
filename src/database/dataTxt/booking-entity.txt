{"id":77,"public_id":"ebYE__oqEFSj0VKL8d8PA","user_id":"NSBo4uU1S0BlGvCcr1Byk","day":"2023-10-25T10:27:07.587Z","desc":"Vulpes ago amaritudo repellendus tepesco quidem. Suscipio viridis quod. Totidem damnatio somniculosus.","updatedAt":"2023-10-25T10:27:07.589Z","createdAt":"2023-10-25T10:27:07.589Z","username":null}