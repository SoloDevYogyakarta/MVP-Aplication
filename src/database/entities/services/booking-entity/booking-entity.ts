import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class BookingEntity extends BaseEntity {
  name: string;
  title: string;
  desc: string;
  user_id: number;
  date: Date;
  time: string;
}

export type BookingCreationAttribute = Optional<BookingEntity, 'id'>;
export interface BookingInstance
  extends Model<BookingCreationAttribute, BookingEntity>,
    BookingEntity {}
export const bookingEntity = sequelize.define<BookingInstance>(
  'BOOKING',
  {
    name: DataTypes.STRING,
    title: DataTypes.TEXT,
    desc: DataTypes.TEXT,
    user_id: DataTypes.INTEGER,
    date: DataTypes.DATE,
    time: DataTypes.STRING,
  } as ModelAttributes<BookingInstance>,
  {
    tableName: 'BOOKING',
    schema: 'SERVICES',
  },
);
