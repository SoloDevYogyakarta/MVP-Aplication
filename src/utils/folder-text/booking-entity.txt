{"id":42,"name":"Shelia Spencer","user_id":1051,"title":"Iure spiritus dolorum acies. Vorax qui casso sulum universe vetus necessitatibus vapulus. Arcus dolore strenuus.","desc":"Est pax coaegresco comptus conventus aeneus. Vindico adduco celer appello concido comes tum vergo tollo. Acquiro votum vestrum theca.","date":"2023-11-06T15:51:07.064Z","time":"22:51:07","updatedAt":"2023-11-06T15:51:07.066Z","createdAt":"2023-11-06T15:51:07.066Z"}