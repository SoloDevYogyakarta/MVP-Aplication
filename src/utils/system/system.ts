import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
