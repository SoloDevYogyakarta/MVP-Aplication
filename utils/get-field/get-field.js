"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map