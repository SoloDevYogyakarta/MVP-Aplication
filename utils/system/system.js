"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removepath = exports.readpath = exports.createpath = exports.joinpath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = require("path");
function joinpath(path) {
    return (0, path_1.join)(__dirname, path);
}
exports.joinpath = joinpath;
function createpath(path, data) {
    if (typeof data !== 'string') {
        data = JSON.stringify(data);
    }
    fs_1.default.writeFileSync(joinpath(path), data);
}
exports.createpath = createpath;
function readpath(path) {
    return fs_1.default.readFileSync(joinpath(path), { encoding: 'utf-8' });
}
exports.readpath = readpath;
function removepath(path) {
    fs_1.default.unlinkSync(joinpath(path));
}
exports.removepath = removepath;
//# sourceMappingURL=system.js.map