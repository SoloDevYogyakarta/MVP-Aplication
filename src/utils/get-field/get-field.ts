import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
