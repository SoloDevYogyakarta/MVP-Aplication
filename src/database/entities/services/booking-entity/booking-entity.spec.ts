import { faker } from '@faker-js/faker';
import { ModelCtor } from 'sequelize';
import { getfield } from '../../../../utils/get-field/get-field';
import { bookingEntity, BookingInstance } from './booking-entity';
import moment from 'moment';
import { createpath } from '../../../../utils/system/system';

describe('BookingEntity', () => {
  let user_id!: number;
  let entity: ModelCtor<BookingInstance>;

  beforeEach(() => {
    entity = bookingEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    user_id = getfield('user-http-entity').id;
  } catch (err) {
    // empty
  }

  if (user_id) {
    it('create', async () => {
      const create = await entity.create({
        name: faker.person.fullName(),
        user_id,
        title: faker.lorem.paragraph(),
        desc: faker.lorem.paragraph(),
        date: new Date(),
        time: moment(new Date()).format('HH:mm:ss'),
      });
      create.save();
      createpath('../folder-text/booking-entity.txt', create);
      expect(create.user_id).toEqual(user_id);
    });
  }
});
