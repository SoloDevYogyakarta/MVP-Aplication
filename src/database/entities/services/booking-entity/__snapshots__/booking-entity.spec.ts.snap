// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingEntity render correctly 1`] = `[Function]`;
